### PR TITLE
Post dictionary API

### LINKED ISSUE #13: Create API to upload dictionary css files

### LINKED ISSUE #21: Create API to allow uploading of dictionary metadata and use it in Webonary
Some of the data that Webonary would need are:
- alphabet, letter heads, sorting sequence, per language
- number of entries
- last update and last upload timestamp

We will need to determine what is possible from FLex, and what can be derived from entries data.

Also, we should consult the user base for other data that can be delivered from API.

### PR DESCRIPTION
**Pull request by hanatgit**
API to post dictionary metadata and get it.

See doc at https://www.hanat.work/wp/wp-content/uploads/api/#api-Dictionary-PostDictionary.

This API is used in Webonary to retrieve and utilize the following:
- css files
- browse alphabet
- last updated timestamp for dictionary
- total number of entries and reverse entries

This PR also includes refactoring for for API paths and renaming from "load" to "post" for clarity.

### CODE REVIEW

File: webonary-cloud-api/lambda/getDictionary.ts
Comment (jasonleenaylor) [inline webonary-cloud-api/lambda/getDictionary.ts:9]: Since this is the exported function it doesn't *need* a more descriptive name, but it could still help maintainers don't you think?

**Review by jasonleenaylor — APPROVED** [review]
So I saw a number of helpful improvements and nothing problematic. I did have one comment which we can consider regarding function names for maintenance sake, but I wouldn't hold this up for it.

File: webonary-cloud-api/lambda/getDictionary.ts
Comment (hanatgit) [reply webonary-cloud-api/lambda/getDictionary.ts:9]: This seems to be a standard with Lambda functions. Also, in the cdk stack.ts code, I can use a standard function using this common naming scheme:

```
function defaultLambdaFunctionProps(functionName: string): lambda.FunctionProps {
  const props: lambda.FunctionProps = {
    functionName,
    runtime: lambda.Runtime.NODEJS_12_X,
    code: new lambda.AssetCode('lambda'),
    handler: `${functionName}.handler`,
    timeout: cdk.Duration.seconds(60),
  };
  return props;
}
```

File: wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php
Comment (phillip-hopper) [inline wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php:280]: I know this line didn't change, but if it's possible for the `$lastEditDate` to be null, then this line should be 
```
if(!empty($lastEditDate) && $lastEditDate != "0000-00-00 00:00:00")
```

File: wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php
Comment (phillip-hopper) [inline wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php:51]: There should be no empty return statements in this function. They should be `return null;` or `return '';`.

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php
Comment (hanatgit) [reply wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php:51]: done :-) thanks.

File: wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php
Comment (hanatgit) [reply wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php:280]: thanks! done.

**Review by phillip-hopper — APPROVED** [review]

**Pull request merged**
